asv_bench/benchmarks/index_object.py

```diff
@@ -258,4 +258,15 @@ def peakmem_gc_instances(self, N):
             gc.enable()
 
 
+class Creation:
```

Comment:
do none of the existing asvs cover this?

Reply:
You are right. it exists in `ctors.py

```python
class SeriesDtypesConstructors:
    def setup(self):
        N = 10**4
        self.arr = np.random.randn(N)
        self.arr_str = np.array(["foo", "bar", "baz"], dtype=object)
        self.s = Series(
            [Timestamp("20110101"), Timestamp("20120101"), Timestamp("20130101")]
            * N
            * 10
        )

    def time_index_from_array_string(self):
        Index(self.arr_str)

    def time_index_from_array_floats(self):
        Index(self.arr)

    def time_dtindex_from_series(self):
        DatetimeIndex(self.s)

    def time_dtindex_from_index_with_series(self):
        Index(self.s)

```

It doesn't hit all the dtypes though. I can expand it following the same naming conventions.